services:
  ctfd:
    image: ctfd/ctfd:3.7.7@sha256:9847758cdafc5ab86bdc121353dcf5a27a29ce313587270ee90a71bfbda2b910
    container_name: ctfd
    ports:
      - 8000:8000
    networks:
      - testing
    volumes:
      - ../:/opt/CTFd/CTFd/plugins/ctfd-chall-manager
    environment:
      LOG_LEVEL: DEBUG
      PLUGIN_SETTINGS_CM_API_URL: http://chall-manager:8080/api/v1
      PLUGIN_SETTINGS_CM_MANA_TOTAL: 10
    depends_on:
      - chall-manager
    healthcheck:
      test: python3 -c 'import requests; requests.get("http://localhost:8000")'
      interval: 10s
      retries: 3
      timeout: 10s

  chall-manager:
    image: ctferio/chall-manager:dev
    container_name: chall-manager
    pull_policy: always
    ports:
      - 8080:8080
    environment:
      SWAGGER: true
      KUBECONFIG: /tmp/config
      OCI_INSECURE: true
    volumes:
      - ~/.kube/config:/tmp/config:ro
    networks:
      - testing

  chall-manager-janitor:
    image: ctferio/chall-manager-janitor:dev
    container_name: chall-manager-janitor
    pull_policy: always
    environment:
      URL: chall-manager:8080
      TICKER: 30s
    networks:
      - testing

  ctfd-setup:
    image: ctferio/ctfd-setup:v1.6.0@sha256:b716c414170b4b17aa317910d8c12e53f2892867791c4912ff3319984a1ea975
    environment:
      URL: http://ctfd:8000
      FILE: /.ctfd.yaml
    volumes:
      - .ctfd.yaml:/.ctfd.yaml
    networks:
      - testing
    depends_on:
      ctfd:
        condition: service_healthy

  registry:
    image: registry:2
    container_name: registry
    ports:
      - 5000:5000
    networks:
      - testing

networks:
  testing:
    driver: bridge
